c/blake3_llfio.cpp

```diff
@@ -0,0 +1,103 @@
+#include <variant>
+
+#include <llfio.hpp>
+
+#include "blake3.h"
+#include "blake3_impl.h"
+
+namespace llfio = LLFIO_V2_NAMESPACE;
+
+template <class... Ts> struct overloads : Ts... {
+  using Ts::operator()...;
+};
+// NOTE: deduction guide only needed for Apple Clang now.
+template <class... Ts> overloads(Ts...) -> overloads<Ts...>;
+
+INLINE auto copy_wide(blake3_hasher *self, llfio::path_view path,
+                      llfio::file_handle &file) noexcept
+    -> llfio::result<std::uint64_t> {
+  std::array<std::byte, 65535> buffer{};
+  std::uint64_t total = 0;
+  while (true) {
+    auto result = llfio::read(file, total, {buffer});
+    // Errors other than "interrupted" are immediately forwarded to caller.
+    if (result.has_error() && result.error() != llfio::errc::interrupted) {
+      return std::move(result).as_failure();
+    }
+    if (result.has_value()) {
+      auto bytes_read = result.value();
+      blake3_hasher_update(self, buffer.data(), bytes_read);
+      total += bytes_read;
+      if (bytes_read == 0) {
+        break;
+      }
+    }
```

Comment:
Non-portable EOF handling see ned14/llfio#29

Reply:
I did see that issue but didn't read the whole thing in detail beyond the first couple comments. I see there's more subtlety involved there so I'll refactor this.